tools/wake/main.cpp

```diff
@@ -176,75 +165,61 @@ void inspect_database(const CommandLineOptions &clo, Database &db, const std::st
     return;
   }
 
-  std::unordered_map<long, JobReflection> captured_jobs = {};
-  Set<long> intersected_job_ids = suniversal<long>();
+  std::vector<std::vector<std::string>> collect_ands = {};
 
-  if (!clo.job_ids.empty()) {
-    intersected_job_ids =
-        apply_inspection_query(captured_jobs, std::move(intersected_job_ids), clo.job_ids,
-                               [&](auto a) { return db.job_ids_matching(a); });
-  }
+  // Process --job
+  make_and_group(clo.job_ids, "cast(job_id as TEXT)", "", collect_ands);
 
-  if (!clo.input_files.empty()) {
-    intersected_job_ids = apply_inspection_query(
-        captured_jobs, std::move(intersected_job_ids), clo.input_files,
-        [&](auto a) { return db.input_files_matching(wcl::make_canonical(wake_cwd + a)); });
-  }
+  // --label
+  make_and_group(clo.labels, "label", "", collect_ands);
 
-  if (!clo.output_files.empty()) {
-    intersected_job_ids = apply_inspection_query(
-        captured_jobs, std::move(intersected_job_ids), clo.output_files,
-        [&](auto a) { return db.output_files_matching(wcl::make_canonical(wake_cwd + a)); });
-  }
+  // --input_files
+  make_and_group(clo.input_files, "input_files", "<d>", collect_ands);
 
-  if (!clo.labels.empty()) {
-    intersected_job_ids =
-        apply_inspection_query(captured_jobs, std::move(intersected_job_ids), clo.labels,
-                               [&](auto a) { return db.labels_matching(a); });
-  }
+  // --output_file
+  make_and_group(clo.output_files, "output_files", "<d>", collect_ands);
 
-  if (!clo.tags.empty()) {
-    intersected_job_ids =
-        apply_inspection_query(captured_jobs, std::move(intersected_job_ids), clo.tags,
-                               [&](auto a) { return db.tags_matching(a); });
-  }
+  // --tag
+  make_and_group(clo.tags, "tags", "<d>", collect_ands);
 
-  if (clo.last_use) {
-    intersected_job_ids =
-        upkeep_intersects(captured_jobs, std::move(intersected_job_ids), db.last_use());
+  // --last-exe
+  if (clo.last_exe) {
+    collect_ands.push_back({"run_id == (select max(run_id) from jobs)"});
+    collect_ands.push_back({"substr(cast(commandline as text),1,1) <> '<'"});
+    collect_ands.push_back({"label <> '<hash>'"});
   }
 
-  if (clo.last_exe) {
-    intersected_job_ids =
-        upkeep_intersects(captured_jobs, std::move(intersected_job_ids), db.last_exe());
+  // --last-use
+  if (clo.last_use) {
+    collect_ands.push_back({"use_id == (select max(run_id) from jobs)"});
+    collect_ands.push_back({"substr(cast(commandline as text),1,1) <> '<'"});
+    collect_ands.push_back({"label <> '<hash>'"});
```

Comment:
Maybe we should factor this pair into a `hide_system_jobs` function or something?